client/app/bundles/course/experience-points/disbursement/components/forms/ForumDisbursementForm.tsx

```diff
@@ -0,0 +1,279 @@
+    );
+    setDialogName(name);
+    setDialogPosts(newPosts);
+  };
+
+  return (
+    <>
+      <form
+        encType="multipart/form-data"
+        id="forum-form"
+        noValidate
+        onSubmit={handleSubmit((data) => {
+          const forumData: ForumDisbursementFormData = { ...filters, ...data };
+          onFormSubmit(forumData);
+        })}
+        style={{ display: forumUsers.length === 0 ? 'none' : 'contents' }}
+      >
+        <ErrorText errors={errors} />
+        <Grid container direction="row" columnSpacing={2} rowSpacing={2}>
+          <Grid item xs>
+            <Controller
+              control={control}
+              name="reason"
+              render={({ field, fieldState }): JSX.Element => (
+                <FormTextField
+                  className="forum_disbursement_reason"
+                  field={field}
+                  fieldState={fieldState}
+                  label={<FormattedMessage {...translations.reason} />}
+                  // @ts-ignore: component is still written in JS
+                  fullWidth
+                  InputLabelProps={{
+                    shrink: true,
+                  }}
+                  required
+                  variant="standard"
+                />
+              )}
+            />
+          </Grid>
+          <Grid item>
+            <Button
+              color="primary"
+              className="forum-btn-submit"
+              disabled={!isDirty}
+              form="forum-form"
+              key="forum-form-submit-button"
+              type="submit"
+              variant="outlined"
+              style={{ marginBottom: '10px', marginTop: '10px' }}
+            >
+              <FormattedMessage {...translations.submit} />
+            </Button>
+          </Grid>
+        </Grid>
+        <ForumDisbursementTable
+          forumUsers={users}
+          pointTextFieldArray={pointTextFieldArray}
+          onPostClick={onPostClick}
+        />
+      </form>
+      <Dialog
+        open={!!dialogName}
+        maxWidth="lg"
+        fullWidth
+        PaperProps={{
+          style: { overflowY: 'inherit' },
+        }}
+        onClose={(): void => {
+          setDialogName('');
+        }}
+      >
+        <DialogTitle
+          borderBottom="1px solid #ccc"
+          style={{
+            display: 'flex',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+            padding: '10px 24px',
+          }}
+        >
+          <div>{`Posts by ${dialogName}`}</div>
+          <IconButton onClick={(): void => setDialogName('')}>
+            <CloseIcon />
+          </IconButton>
+        </DialogTitle>
+        <DialogContent style={{ height: '70vh' }}>
+          <ForumPostTable posts={dialogPosts} />
+        </DialogContent>
+      </Dialog>
+    </>
+  );
+};
+
+export default memo(
+  injectIntl(ForumDisbursementForm),
+  (prevProps, nextProps) => {
+    const bool =
```

Comment:
You can use equal from fast-deep-equal instead. Check other part of the codebase